### PR TITLE
Add User Endpoints

### PR DESCRIPTION
**Pull request by Matthew-Grayson**
#  #

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: backend/src/xfd_django/xfd_api/api_methods/user.py [hunk 1]
```diff
@@ -3,30 +3,482 @@
...
+        current_user.dateAcceptedTerms = datetime.now()
+        current_user.acceptedTermsVersion = version
+        current_user.save()
+
+        return UserSchema.model_validate(current_user)
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
+
+
+#  TODO: Add user context and permissions
```
Comment (cduhn17) [inline backend/src/xfd_django/xfd_api/api_methods/user.py]: Create ticket associated with all TODO messages for tracking purposes

File: backend/src/xfd_django/xfd_api/api_methods/user.py [hunk 2]
```diff
@@ -3,30 +3,482 @@
...
+        raise HTTPException(status_code=500, detail=f"Failed to send email: {str(e)}")
+
+    return {"statusCode": 200, "body": "User registration approved."}
+
+
+def deny_user_registration(user_id: str, current_user: User):
+    """Deny a user's registration by user ID."""
+    try:
+        # Validate UUID format for the user_id
+        if not is_valid_uuid(user_id):
```
Comment (cduhn17) [inline backend/src/xfd_django/xfd_api/api_methods/user.py]: For consistency for logic checks on `is_valid_uuid` should be outside of try blocks like shown on `def user_register_approve`. In general anything that does not involve i.e. database connections or services connections like email(this is not an exhaustive list) should be outside of try blocks. Things that qualify for in try block are exception prone operations.

File: backend/src/xfd_django/xfd_api/helpers/email.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+from fastapi import HTTPException
+from jinja2 import Template
+
+from .s3_client import S3Client
+
+
+def send_invite_email(email, organization=None):
+    """Send an invitation email to the specified address."""
+    frontend_domain = os.getenv("FRONTEND_DOMAIN")
+    reply_to = os.getenv("CROSSFEED_SUPPORT_EMAIL_REPLYTO")
```
Comment (cduhn17) [inline backend/src/xfd_django/xfd_api/helpers/email.py]: if you want to drive constants from the settings file
`import django.conf import settings`. Then you can change os.getenv to `settings.("CROSSFEED_SUPPORT_EMAIL_REPLYTO")`. The reasoning is then constants wont be overridden by test mocks and to not depend on the environment to get constants. Django caches constants for efficient retrieval as well on start up.

**Review by cduhn17 — COMMENTED** [review]
See comments.

**Review by nickviola — APPROVED** [review]
I'm not sure if we still need the invitation email stuff, but we can address later in a SPIKE ticket. Otherwise this LGTM

**Review by aloftus23 — APPROVED** [review]
Looks good to me

**Pull request merged**